frontend/src/utils/websockets.js

```diff
@@ -43,25 +44,33 @@ export function connectWS(topic, onMessageCallback) {
 
   const connect = () => {
     let reconnectCounter = 0;
+    let badGatewayStatus = false;
     this.sockClient = new Client({
       brokerURL: config.WS_URL,
       debug(strMessage) {
         // console.log('debug: ', strMessage);
         // -- detect reconnect and increment the reconnect counter
         if (strMessage.includes('reconnect')) {
           getUserSession()
-            .then(({ data }) => {
+            .then(({ data, status }) => {
               reconnectCounter =
                 data && !data.loggedIn ? reconnectCounter + 1 : 0;
+
+              if (status === 502) badGatewayStatus = true;
             })
             .catch(() => {
               reconnectCounter += 1;
             });
         }
+        // update the store flag
+        badGatewayStatus &&
+          store.dispatch(noConnection({ errorType: BAD_GATEWAY_ERROR }));
+
         // -- if more than max allowed reconnect times  ->  deactivate
         if (reconnectCounter > maxReconnectTimesNo) {
           this.reconnectDelay = 0; // 0 - deactivates the sockClient
-          store.dispatch(noConnection(true));
+          !badGatewayStatus &&
+            store.dispatch(noConnection({ errorType: NO_CONNECTION_ERROR }));
```

Comment:
why do we have to treat NO_CONNECTION_ERROR and BAD_GATEWAY_ERROR differently?

Reply:
I thought we want separate messages for each case

Reply:
i mean, errors might be different but we need to have the same approach like retry etc etc

Reply:
ok. got it